pkg/controller/resourceclaim/controller.go

```diff
@@ -354,15 +354,48 @@ func (ec *Controller) podNeedsWork(pod *v1.Pod) (bool, string) {
 	return false, "nothing to do"
 }
 
-func (ec *Controller) enqueueResourceClaim(logger klog.Logger, obj interface{}, deleted bool) {
-	if d, ok := obj.(cache.DeletedFinalStateUnknown); ok {
-		obj = d.Obj
+func (ec *Controller) enqueueResourceClaim(logger klog.Logger, oldObj, newObj interface{}) {
+	deleted := newObj != nil
```

Comment:
I think you have a potential panic here. Should you check if oldObj is not nil?

Otherwise, I see a lot of potential failures.

Reply:
Trying to cast a nil interface is okay. It sets `ok` to false.